### PR TITLE
fix(external_services): add proto build under the dynamic_routing feature flag

### PR DESCRIPTION
**Pull request by Aprabhat19**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description
add proto build under the dynamic_routing feature flag. Now the application would run without having to install protoc

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## How did you test it?

Cannot be tested as it is for compiling the application without having to mandatorily install proton

## Checklist

- [ ] I formatted the code `cargo +nightly fmt --all`
- [ ] I addressed lints thrown by `cargo clippy`
- [ ] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

**Review by Chethan-rao — APPROVED** [review]

File: crates/external_services/Cargo.toml [hunk 1]
```diff
@@ -49,7 +49,8 @@ api_models = { version = "0.1.0", path = "../api_models", optional = true }
 
 
 [build-dependencies]
-tonic-build = "0.12"
+tonic-build = { version =  "0.12" , optional = true }
+router_env = { version = "0.1.0", path = "../router_env", default-features = false  }
```
Comment (SanchithHegde) [inline crates/external_services/Cargo.toml:53]: For now, this dependency can also be optional, right? Behind the `dynamic_routing` feature flag perhaps?

Comment (Aprabhat19) [reply crates/external_services/Cargo.toml:53]: Just the build dependency right?

File: crates/external_services/build.rs [hunk 2]
```diff
@@ -1,15 +1,13 @@
...
-    // Compile the .proto file
-    tonic_build::compile_protos(proto_file).expect("Failed to compile protos ");
+    #[cfg(feature = "dynamic_routing")]
+    {
+        // Get the directory of the current crate
+        let proto_file = router_env::workspace_path()
+            .join("proto")
+            .join("success_rate.proto");
+        // Compile the .proto file
+        tonic_build::compile_protos(proto_file).expect("Failed to compile protos ");
```
Comment (Chethan-rao) [inline crates/external_services/build.rs:10]:
```suggestion
        tonic_build::compile_protos(proto_file).expect("Failed to compile success_rate proto file");
```

**Review by Chethan-rao — CHANGES_REQUESTED** [review]
Could you also add the grpc config in example toml file

File: config/config.example.toml [hunk 3]
```diff
@@ -745,4 +745,8 @@ delete_token_url= ""          # base url to delete token from token service
 check_token_status_url= ""    # base url to check token status from token service
 
 [network_tokenization_supported_connectors]
-connector_list = "cybersource" # Supported connectors for network tokenization
\ No newline at end of file
+connector_list = "cybersource" # Supported connectors for network tokenization
+
+[grpc_client.dynamic_routing]
+host = "localhost" 
+port = 7000
```
Comment (SanchithHegde) [inline config/config.example.toml:752]: Please add documentation comments for this. Please do the same changes in `config/deployments/env_specific.toml` as well, if not done already.

**Review by SanchithHegde — APPROVED** [review]

File: .github/workflows/CI-pr.yml [hunk 4]
```diff
@@ -122,7 +122,9 @@ jobs:
           toolchain: "${{ env.rust_version }}"
 
       - name: Install Protoc
-        uses: arduino/setup-protoc@v3
+        uses: arduino/setup-protoc@v3with
```
Comment (SanchithHegde) [inline .github/workflows/CI-pr.yml:125]:
```suggestion
        uses: arduino/setup-protoc@v3
```

**Review by SanchithHegde — APPROVED** [review]

**Review by Chethan-rao — APPROVED** [review]

File: docs/try_local_system.md [hunk 5]
```diff
@@ -442,6 +443,13 @@ You can opt to use your favorite package manager instead.
    cargo install just
    ```
 
+6. Install the `protoc` to compile the proto files. (optional: if gRPC services are enabled)
+   
+   ```shell
+   brew install protoc
+   brew install protobuf
+   ```
```
Comment (SanchithHegde) [inline docs/try_local_system.md:451]: 1. I don't think it's optional: we'll need `protoc` to run `cargo check` as well I believe.
2. There's no package called `protoc`, it's only `protobuf`. Keep only the second command.
3. Can you update the instructions for Ubuntu, WSL2 and Windows as well?

Comment (Aprabhat19) [reply docs/try_local_system.md:451]: It would be optional and under a feature flag

**Review by Chethan-rao — APPROVED** [review]

**Review by SanchithHegde — APPROVED** [review]

**Pull request merged**